lib/cadet/course/course.ex

```diff
@@ -57,6 +56,28 @@ defmodule Cadet.Course do
     end
   end
 
+  @doc """
+  Create group entity with given name
+  """
+  def create_group(name, leader, mentor) do
+    %Group{}
+    |> Group.changeset(%{name: name})
+    |> change(%{leader_id: Map.get(leader, :id)})
```

Comment:
This should be part of the changeset itself. You will need to change `Group.changeset/2` as well.